flowbench/CMakeLists.txt

```diff
@@ -2,36 +2,49 @@ project (flowbench)
 
 fdb_find_sources(FLOWBENCH_SRCS)
 
-# include the configurations from benchmark.cmake
-configure_file(benchmark.cmake googlebenchmark-download/CMakeLists.txt)
-# executing the configuration step
-execute_process(
-    COMMAND ${CMAKE_COMMAND} -G "${CMAKE_GENERATOR}" .
-    RESULT_VARIABLE results
-    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/googlebenchmark-download
-)
-# checking if the configuration step passed
-if(results)
-    message(FATAL_ERROR "Configuration step for Benchmark has Failed. ${results}")
-endif()
-# executing the build step
-execute_process(
-    COMMAND ${CMAKE_COMMAND} --build . --config Release
-    RESULT_VARIABLE results
-    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/googlebenchmark-download
-)
-# checking if the build step passed
-if(results)
-  message(FATAL_ERROR "Build step for Benchmark has Failed. ${results}")
+# There is no good way to incorporate the recommended googlebenchmark download + build
+# process with one that checks to see if googlebenchmark has already been downloaded
+# and built.
+if(EXISTS /opt/googlebenchmark-f91b6b)
```

Comment:
I suppose if we use

```
find_package(GoogleBenchmark REQUIRED)
```

it might be more CMake-ic. I noticed that there is no GoogleBenchmark find_package in CMake but I suppose

```
https://code.qt.io/cgit/qt-creator/qt-creator.git/tree/cmake/FindGoogleBenchmark.cmake
```

might be useful. Yet the concern is our code is Apache licensed while Qt-Creator's code is GPLv3 licensed.

Reply:
Using GPL code is fine, we just need to retain the copyright in the file. There are already some GPL code in the repo, e.g., libeio.

Reply:
That seems like a fine fix for the else branch (which I didn't write, and which came from upstream), since it is much cleaner than the pre-existing logic.

I took a look at it, and (for the case where we want to use existing binaries / headers in /opt) we'd want to eliminate the default paths in the HINTS list, and also the logic that auto-compiles googlebenchmark.  So, for that use case, the new cmake file isn't doing much.

However, the whole point of this PR is to make the build break if the /opt directory for some dependency exists but cannot be used.   I don't want to transitively fix the cmake build definitions (that we're explicitly trying to disable in CI!) as part of this work.

Reply:
> Using GPL code is fine, we just need to retain the copyright in the file. There are already some GPL code in the repo, e.g., libeio.

I am not certain about this. libeio is under GPL v2; while the qt-creator code is dual-licensed and the license we may use is GPL v3, which is incompatible with our ApacheV2 license, see, e.g.

https://www.apache.org/licenses/GPL-compatibility.html